### PR TITLE
[13.0][FIX] hr_employee_calendar_planning: Prevent disable resource.calendar when is related to some employee

### PR DESCRIPTION
**Pull request by victoralmau**
Prevent disable `resource.calendar` when is related to some employee.

It's need to change in 14.0 too.

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT31007

### CODE REVIEW

File: hr_employee_calendar_planning/models/resource_calendar.py [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+    def _check_active(self):
+        for item in self:
+            total_items = self.env["hr.employee.calendar"].search_count(
+                [("calendar_id", "=", item.id)]
+            )
+            if total_items > 0:
+                raise ValidationError(
+                    _(
+                        "%s is related with some employee in calendar planing,"
+                        "you should change it before." % item.name
```
Comment (joao-p-marques) [inline hr_employee_calendar_planning/models/resource_calendar.py]:
```suggestion
                        "%s is related to %s employee(s) in calendar planing,"
                        " you should change it first." % (item.name, total_items)
```

**Review by joao-p-marques — COMMENTED** [review]
Code and functional review :+1:

Just a little suggestion on the error message

**Review by joao-p-marques — APPROVED** [review]

File: hr_employee_calendar_planning/models/resource_calendar.py [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
 
     active = fields.Boolean(default=True)
 
+    @api.constrains("active")
+    def _check_active(self):
+        for item in self:
+            total_items = self.env["hr.employee.calendar"].search_count(
+                [("calendar_id", "=", item.id)]
+            )
+            if total_items > 0:
```
Comment (pedrobaeza) [inline hr_employee_calendar_planning/models/resource_calendar.py]:
```suggestion
            if total_items:
```

File: hr_employee_calendar_planning/models/resource_calendar.py [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+                raise ValidationError(
+                    _(
+                        "%s is related to %s employee(s) in calendar planing,"
+                        " you should change it first." % (item.name, total_items)
+                    )
```
Comment (pedrobaeza) [inline hr_employee_calendar_planning/models/resource_calendar.py]: Never include the placeholder elements inside the call to translate

```suggestion
                    _(
                        "%s is used in %s employee(s). You should change them first."
                    ) % (item.name, total_items)
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**